app/services/users/update.rb

```diff
@@ -24,7 +24,7 @@ def self.call(user, updated_attributes = {})
 
     def initialize(user, updated_attributes)
       @user = user
-      @profile = user.profile
+      @profile = user.profile || user.create_profile
```

Comment:
I understand and agree with the purpose for this PR, but it's strange: in the User [model](https://github.com/forem/forem/blob/fa093b0a92e137d06f6718ebca843eb78bad8f39/app/models/user.rb#L20), we create the user profile in an `after_create_commit`:

```ruby
after_create_commit -> { Profile.create(user: self) }, unless: :_skip_creating_profile
```

While working on UsersSettings, I remember encountering this same problem of the profile not being created for a new user (in some instances). Seems like it's still persisting. 😟 

Anyways, good job Jamie. I just wanted to surface some of my experience with the issue. 👍🏾

Reply:
@msarit I really appreciate this insight. I noticed the callback, too, and I wonder if there's a race condition or something somewhere or if maybe the `Profile.create` failed silently? I'm not suggesting that's what it is, just that those are the only reasons I can imagine that `profile` could be `nil` when this line from the diff is executed. On the bright side, if it's a race condition, we have [a uniqueness constraint](https://github.com/forem/forem/blob/fa093b0a92e137d06f6718ebca843eb78bad8f39/db/schema.rb#L974) that keeps us from creating a second profile for the user.

Do you have context for why it's an `after_create_commit` instead of an `after_create`?

Reply:
@jgaskins if I remember correctly, it was to ensure that the profile is created after the database transaction for the user-create has completed.

Reply:
@jgaskins This happens every so often, but never often enough that we actually bothered to investigate it. Though one of the last times we definitely discussed changing the callback.